File: src/Entity/Poste.php
<?php

namespace App\Entity;

use App\Repository\PosteRepository;
use Doctrine\ORM\Mapping as ORM;
use symfony\Component\Serializer\annotaion\groups;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=PosteRepository::class)
 */
class Poste
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="tapez votre titre")
     * @Assert\Regex(
    pattern="/\d/",
    match=false,
    message="ton titre ne doit pas contenir un numero")
     */
    private $titre;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="tapez votre sujet")
     */
    private $sujet;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="upload votre image")
     * @Assert\File(mimeTypes={ "image/jpeg" })
     */
    private $image;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="choisir groupe")
     */
    private $groupe;

    /**
     * @ORM\OneToMany(targetEntity=Commentaire::class, mappedBy="id_poste")
     */
    private $commentaires;

    /**
     * @ORM\OneToMany(targetEntity=Like::class, mappedBy="id_poste")
     */
    private $likes;

    public function __construct()
    {
        $this->commentaires = new ArrayCollection();
        $this->likes = new ArrayCollection();
    }





    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getSujet(): ?string
    {
        return $this->sujet;
    }

    public function setSujet(string $sujet): self
    {
        $this->sujet = $sujet;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage( $image)
    {
        $this->image = $image;

        return $this;
    }

    public function getGroupe(): ?string
    {
        return $this->groupe;
    }

    public function setGroupe(string $groupe): self
    {
        $this->groupe = $groupe;

        return $this;
    }

    /**
     * @return Collection|Commentaire[]
     */
    public function getCommentaires(): Collection
    {
        return $this->commentaires;
    }

    public function addCommentaire(Commentaire $commentaire): self
    {
        if (!$this->commentaires->contains($commentaire)) {
            $this->commentaires[] = $commentaire;
            $commentaire->setIdPoste($this);
        }

        return $this;
    }

    public function removeCommentaire(Commentaire $commentaire): self
    {
        if ($this->commentaires->removeElement($commentaire)) {
            // set the owning side to null (unless already changed)
            if ($commentaire->getIdPoste() === $this) {
                $commentaire->setIdPoste(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection|Like[]
     */
    public function getLikes(): Collection
    {
        return $this->likes;
    }

    public function addLike(Like $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setIdPoste($this);
        }

        return $this;
    }

    public function removeLike(Like $like): self
    {
        if ($this->likes->removeElement($like)) {
            // set the owning side to null (unless already changed)
            if ($like->getIdPoste() === $this) {
                $like->setIdPoste(null);
            }
        }

        return $this;
    }



}

File: SprintMobile/src/com/mycomany/entities/Poste.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

/**
 *
 * @author User
 */
public class Poste {
    private int id;
    private String titre,sujet,image,groupe;
    public Poste(){}

    public Poste(int id, String titre, String sujet, String image, String groupe) {
        this.id = id;
        this.titre = titre;
        this.sujet = sujet;
        this.image = image;
        this.groupe = groupe;
    }

    public Poste(String titre, String sujet, String image, String groupe) {
        this.titre = titre;
        this.sujet = sujet;
        this.image = image;
        this.groupe = groupe;
    }

    public int getId() {
        return id;
    }

    public String getTitre() {
        return titre;
    }

    public String getSujet() {
        return sujet;
    }

    public String getImage() {
        return image;
    }

    public String getGroupe() {
        return groupe;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public void setSujet(String sujet) {
        this.sujet = sujet;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setGroupe(String groupe) {
        this.groupe = groupe;
    }

    @Override
    public String toString() {
        return "Poste{" + "titre=" + titre + ", sujet=" + sujet + ", image=" + image + ", groupe=" + groupe + '}';
    }
    
    
    
}
